### ISSUE TITLE
Double-quoted dates may not be recognized in MySQL dialect

### ISSUE THREAD

**Issue by rolandwalker** [tobymao/sqlglot#375]
per https://github.com/dbcli/mycli/pull/1075#discussion_r957254586

This expression 
```
python -c 'import sqlglot; print(sqlglot.parse_one("""SELECT "2021-01-01" + interval 1 month""", read="mysql").sql(dialect="mysql"))'
```
throws 
```
sqlglot.errors.ParseError: Invalid expression / Unexpected token. Line 1, Col: 21.
  SELECT "2021-01-01" + interval 1 month
```

whereas single-quoting the date is accepted.

This would seem to indicate that the string is being read in a strange way:
```
$ python -c 'import sqlglot; print(sqlglot.parse_one("""SELECT "2021-01-01" """, read="mysql").sql(dialect="mysql"))'
SELECT "2021 - 01 - 01 AS "
```

Thanks for an awesome project.

**Comment by barakalon** [comment]
This should fix the issue: https://github.com/tobymao/sqlglot/pull/376

That will parse string literals with double quotes, always generate SQL with single quotes. I think that's acceptable for your prettify use case. Let me know if otherwise.

**Comment by barakalon** [comment]
Fix available in [v5.1.3](https://pypi.org/project/sqlglot/5.1.3/)

**Comment by rolandwalker** [comment]
> That will parse string literals with double quotes, but it will always generate SQL with single quotes. I think that's acceptable for your prettify use case. Let me know if otherwise.

Perfect.
